src/components/AddressSummaryCard.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components'
+
+import { Address } from '../contexts/addresses'
+import Amount from './Amount'
+import ClipboardButton from './Buttons/ClipboardButton'
+import QRCodeButton from './Buttons/QRCodeButton'
+import Label from './Label'
+
+interface AddressSummaryCardProps {
+  address: Address
+  clickable?: boolean
+  className?: string
+}
+
+const AddressSummaryCard = ({ address, clickable, className }: AddressSummaryCardProps) => {
+  const history = useHistory()
+
+  return (
+    <div className={className}>
+      <ClickableArea
+        onClick={() => clickable && history.push(`/wallet/addresses/${address.hash}`)}
+        clickable={clickable}
+      >
+        <AddressNameSection>
+          <Label color={address.settings.color} truncate>
+            {address.labelDisplay()}
```

Comment:
I agree that the method I've been using so far is not the best. And the proposal to have `get...` functions makes total sense. However, I'd like to propose using computer properties instead of functions, like I do for `availableBalance` for example (instead of creating a function called `getAvailableBalance`. I'll explain in a small commit.

Reply:
done